### ISSUE TITLE
conditional query false positive

### ISSUE THREAD

**Issue by staabm** [staabm/phpstan-dba#319]
in conditons like

```
        $select = 'SELECT `watchlist`.* FROM `watchlist` WHERE (send = ?) AND (kindid = ?)';
        $values = [0, $this->determineKindid()];

        $adaFilter = '';

        $userSession = ClxProductNet_UserSession::getInstance();
        if (is_object($userSession->ada)) {
            $adaFilter = ' AND (adaid = ?)';
            $values[] = $userSession->ada->getAdaid();
        }

        $sessionFilter = ' AND (sessionid = ?)';
        $values[] = ClxProductNet_SessionNS::getId();

        $select .= $adaFilter.$sessionFilter;

        return $this->fetchRowByStatement(new ClxProductNet_DbStatement($select, $values));
```

we see errors like `Query expects 2 placeholders, but 3-4 values are given.`
-> the query expect 3-4 placeholders

**Comment by xPaw** [comment]
A simpler example:

```php
	$q = 'SELECT 1 FROM table WHERE';

	if( true )
	{
		$q .= ' 1=1';
	}

	$pdo->query( $q );
```

> Query error: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL/MariaDB server version for the right syntax to use near 'LIMIT 0' at line 1 (42000).

It simulates `WHERE LIMIT 0`

**Comment by staabm** [comment]
> a simpler example is

On plain phpstan, the query is properly built

https://phpstan.org/r/11f6f7e6-6eec-4cc7-8700-77b7df3e5814

Need to investigate
